### ISSUE TITLE
incorrect location tags for some photos

### ISSUE THREAD

**Issue by erkcan** [jensb/iphoto2xmp#5]
It appears that some photos have incorrect locations entered into their xmp files. Could it be that the code does not clean up the variable that holds the location string after handling each photo and hence photos with no location assigned accidentally get the location data from a previously handled photo?

**Comment by jensb** [comment]
I don't think so (the variable is reinitialized after every loop) but I'll have a look. Are the locations completely nonsensical or just off by some bits? Do they match other locations that you use in your library?

**Comment by erkcan** [comment]
I could not identify a clear pattern. It is just that for some no-location-data photos, the relevant xmp files have location attributes and the values seem to be copies of the locations of some other photos.

**Comment by jensb** [comment]
Can you identify some such photos and send me (privately) the contents of your iPhoto Library's "Database/apdb" subdirectory? I would like to look whether the IP data of these pictures is in the DB. 
Alternatively, could you open a Terminal (if you have sqlite3 installed) and issue this command (edit the path as appropriate):

```
$ sqlite3 ~/Pictures/iPhoto\ Library/Database/apdb/Library.apdb
sqlite> SELECT v.masterId, v.name, v.exifLatitude, v.exifLongitude FROM RKVersion v WHERE v.uuid='UUID' OR v.masterUuid='UUID';
```

The string "UUID" is to be replaced by the UUID of the image, which you can get by looking at the subdirectory of its "Preview", the path looks something like

  ..../Previews/2015/04/27/20150427-150316/ **hJXyMAkk19240fdkas9** /file.jpg

That's the image UUID. The SQL command should output something like this:

```
1|MoreFaces|50.685293|7.264409
```

Image ID, Image Caption, and lat/lng values if present. Post them here. Are those the values that are written into the XMP files? Thanks!

**Comment by erkcan** [comment]
Is there another way to get the UUID? I very rarely modify my photos in iPhoto, so most of them are in original form. For example, there are more than 2400 jpeg files in my Masters directory, while the number of jpegs in my Previews directory is not even 200.

**Comment by erkcan** [comment]
One more question: The SQL command that you want me to run appears to be asking the latitude and longitude from the exif data. I don't think photos for which the location data is in the exif have problems. Actually, now as I look at the xmp file of a problematic photo, I see the following block:

```
   exif:DateTimeOriginal="2011-07-23T22:25:18+03:00"
   exif:ExifVersion="0221"
   exif:GPSLatitudeRef="N"
   exif:GPSLongitudeRef="E"
   exif:GPSVersionID="2.0.0.0"
   video:DateTimeOriginal="2011-07-23T22:25:18+03:00"
   video:ModificationDate="2011-07-23T22:25:18+03:00"
   photoshop:City=""
   photoshop:State=""
   photoshop:Country=""
   Iptc4xmpCore:Location="Nevali Hotel"
   Iptc4xmpCore:CountryCode=""
```

As you can see there is no location data in the exif part of the xmp, just the Iptc part; and that location is incorrect. This file seems to have inherited a location from a photo in another album. That location is one that I had manually entered.

**Comment by jensb** [comment]
You're right. I thought you meant GPS data. Fixed by 2285e3c35.

**Comment by erkcan** [comment]
excellent, thanks.
